### PR TITLE
[runtimes] Fix parsing of LIB{CXX,CXXABI,UNWIND}_TEST_PARAMS

### PR DESCRIPTION
**Pull request by arichardson**
Since 78d649a417b48cb8a2ba2e755f0e7c8fb8b1bb83 the recommended way to pass an executor is to use the _TEST_PARAMS variable, which means we now pass more complicated value (including ones that may contain multiple `=`) as part of this variable. However, the `REGEX REPLACE` being used has greedy matches so everything up to the last = becomes part of the variable name which results in invalid syntax in the generated lit config file.

This was noticed due to builder failures for those using the CrossWinToARMLinux.cmake cache file.

### CODE REVIEW

**Comment by arichardson** [comment]
@vvereschaka I tested this locally by adding a parameter with quotes to my CMake invocation, so it should hopefully work. Could you please test if this fixes the builder?

File: runtimes/cmake/Modules/HandleLitArguments.cmake
Comment (ldionne) [inline runtimes/cmake/Modules/HandleLitArguments.cmake:3]: This looks reasonable, but could we avoid using the global variable `SERIALIZED_LIT_PARAMS` here? We should thread it through the functions instead, otherwise these functions all "return their result" by modifying the global variable, which is yucky.

**Review by ldionne — CHANGES_REQUESTED** [review]

File: clang/cmake/caches/CrossWinToARMLinux.cmake
Comment (vvereschaka) [inline clang/cmake/caches/CrossWinToARMLinux.cmake:157]: It is not working. This will store the empty variables in the cache file. If you check your local `CMakeCache.txt` for these variables you'll find them empty. You need to cache them with the values at the end of preparation (please check my original diff file).

**Comment by vvereschaka** [comment]
@arichardson,
any progress?!

**Comment by arichardson** [comment]
> @arichardson, any progress?!

Sorry thought I'd pushed the latest version earlier, turns out there was an error during the push. Updated now.

File: clang/cmake/caches/CrossWinToARMLinux.cmake
Comment (arichardson) [reply clang/cmake/caches/CrossWinToARMLinux.cmake:157]: Should be fixed in the latest version.

**Comment by vvereschaka** [comment]
@arichardson,
thank you. I'm going to test it during the next 15-20 minutes.

**Comment by vvereschaka** [comment]
@arichardson ,
works fine on the builders locally. Thank you.
Would yo merge these changes if they are ready?

**Comment by arichardson** [comment]
> @arichardson , works fine on the builders locally. Thank you. Would yo merge these changes if they are ready?

Thanks for testing! I'll merge these changes as soon as @ldionne is happy with the new macros. Once that happens I'll push the first commit manually and then merge the second one via the github UI (since we can't do "rebase and merge" due to repo settings).

**Review by vvereschaka — APPROVED** [review]

**Comment by vvereschaka** [comment]
@ldionne @arichardson any progress?

**Comment by vvereschaka** [comment]
@ldionne @arichardson ,
the problem is still there for more than 10 days already. Do you guys plan to fix it!

**Comment by arichardson** [comment]
> @ldionne @arichardson , the problem is still there for more than 10 days already. Do you guys plan to fix it!

I am waiting for @ldionne to approve this PR.

**Comment by vvereschaka** [comment]
@arichardson ,
would you commit these changes without @ldionne 's approval? or revert your #66545?

**Comment by ldionne** [comment]
Thanks for the fix and sorry for the delay!

**Pull request merged**
